modules/Bio/EnsEMBL/Variation/VCFCollection.pm

```diff
@@ -777,6 +779,140 @@ sub get_all_Alleles_by_VariationFeature {
   return \@alleles;
 }
 
+=head2 get_all_PopulationGenotypes_by_VariationFeature
+
+  Arg[1]     : Bio::EnsEMBL::Variation::VariationFeature $vf
+  Arg[2]     : (optional) Bio::EnsEMBL::Variation::Population $pop
+  Arg[3]     : (optional) Bio::EnsEMBL::IO::Parser::VCF4Tabix $vcf
+  Example    : my $population_genotypes = $collection->get_all_PopulationGenotypes_by_VariationFeature($vf)
+  Description: Get all PopulationGenotypes for a given VariationFeature object
+               PopulationGenotype counts and frequencies are only retrieved for ESP from the info field. 
+  Returntype : arrayref of Bio::EnsEMBL::Variation::PopulationGenotype
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub get_all_PopulationGenotypes_by_VariationFeature {
+  my $self = shift;
+  my $vf = shift;
+  my $given_pop = shift;
+  my $vcf = shift;
+
+  assert_ref($vf, 'Bio::EnsEMBL::Variation::VariationFeature');
+  assert_ref($given_pop,'Bio::EnsEMBL::Variation::Population') if (defined($given_pop));
+
+  # if given $vcf, we don't want to call next() on it
+  # as it is a frozen copy created in VCFVF object creation
+  my $next_ok = 1;
+
+  if(!$vcf) {
+    # seek to record for VariationFeature
+    return [] unless $self->_seek_by_VariationFeature($vf);    
+    $vcf = $self->_current();
+  }
+  else {
+    $self->_current($vcf);
+    $next_ok = 0;
+  }
+  # get data from VF
+  my $vf_ref = $vf->ref_allele_string;
+  my $vf_alts = $vf->alt_alleles;
+  # check initial VCF match
+  my $matched = $self->_get_matched_alleles_VF_VCF($vf, $vcf);
+  return [] unless @$matched;
+
+  my @pops = @{$self->get_all_Populations};
+  @pops = grep {$_->name eq $given_pop->name || ($_->{_raw_name} || '') eq $given_pop->name} @pops if $given_pop;
+
+  my $variation = $vf->variation;
+  my $population_genotype_adaptor = $self->use_db ? $self->adaptor->db->get_PopulationGenotypeAdaptor : undef;
+
+  my @population_genotypes = ();
+
+  while(1) {
+    my $info = $vcf->get_info;
+    my $vcf_alts = $vcf->get_alternatives;
+    my @alt_alleles = @$vcf_alts;
+    my $ref = $vcf->get_reference;
+    my %allele_map = map {$_->{b_index} => $_->{a_allele}} @$matched;
+    my @sorted_gts = ();
+    my $info_gts = $info->{'GTS'};
+    if (defined $info_gts) {
+      my @gts = split(',', $info_gts);
+      foreach my $gt (@gts) {
+        my @tmp_gt = ();
+        if ($gt =~ /^[ACGT]+$/i) {
+          my @gt_alleles = split('', $gt);
+          foreach my $gt_allele (@gt_alleles) {
+            push @tmp_gt, $gt_allele;
+          }
+        } elsif ($gt =~ /[A\d|R]+/) { #Observed Genotypes. For INDELs, A1, A2, or An refers to the N-th alternate allele while R refers to the reference allele.
```

Comment:
Suggest adding test for INDELs. The code is called in test in populationGenotypeAdaptor.t

Reply:
I'm testing for an insertion which covers the elsif block.

Reply:
added also test for indel